src/containers/Redemptions.jsx

```diff
@@ -220,19 +214,39 @@ class Redemptions extends React.Component {
 
   /**
    * @name handleClick
-   * @param {Boolean} isApproved whether or not a redemption request has been approved
+   * @param {Boolean} clickAction which button has been clicked
    * @param {String} redemptionId id of clicked redemption request
    * @summary calls verifyRedemption action creator when approve or reject button is clicked
    */
-  handleClick = (isApproved, redemptionId) => {
-    if (!isApproved) {
-      const redemption = this.state.filteredActivities.find(r => r.id === redemptionId);
-      this.toggleOpenModal(redemption);
-    } else {
-      this.props.verifyRedemption(redemptionId, isApproved);
+  handleClick = (clickAction, redemptionId) => {
+    const { APPROVE, REJECT, MORE_INFO } = clickActions;
+    if (clickAction === REJECT || clickAction === MORE_INFO) {
+      const selectedRedemption = this.state.filteredActivities.find(r => r.id === redemptionId);
+      selectedRedemption.rejectClicked = clickAction === REJECT;
+      this.setState({
+        showModal: true,
+        selectedRedemption,
+      });
+    } else if (clickAction === APPROVE) {
+      this.props.verifyRedemption(redemptionId, clickAction);
     }
   }
 
+  handleClickToEdit = (redemptionId) => {
```

Comment:
I noticed that the contents of this function are similar to that of `handleClick, line 223-228`,  could you shed more light on this?

Reply:
`handleClickToEdit` is called when the user clicks on the clickable area of an activity card. It sets state with the clicked on redemption by finding it using the id from all the redemptions, and also opens the modal by setting show modal to true. `handleClick` is called when the action buttons (approve, reject, comment) are clicked and calls the verification thunk